src/main/scala/util/Annotations.scala

```diff
@@ -66,11 +70,28 @@ case class TopLevelPortAnnotation(
     protocol: String,
     tags: Seq[String],
     mapping: Seq[AddressMapEntry]) extends SingleTargetAnnotation[ComponentName] {
-  def duplicate(n: ComponentName) = this.copy(n)
+  def duplicate(n: ComponentName): TopLevelPortAnnotation = this.copy(n)
+}
+
+/** Record the resetVector. */
+case class ResetVectorAnnotation(target: Named, resetVec: BigInt) extends SingleTargetAnnotation[Named] {
+  def duplicate(n: Named): ResetVectorAnnotation = this.copy(n)
 }
 
 /** Helper object containing methods for applying annotations to targets */
-object annotated {
+object Annotated {
+  def interrupts(component: InstanceId, name: String, interrupts: Seq[Int]): Unit = {
+    annotate(new ChiselAnnotation {def toFirrtl: Annotation = InterruptsPortAnnotation(
+      component.toNamed,
+      name,
+      interrupts
+    )})
+  }
+
+  def resetVector(component: InstanceId, resetVec: BigInt): Unit = {
```

Comment:
What do you think about having this function do the assignment as well:
` component := resetVec.U`
Seem like it will simplify using it downstream?

Reply:
Lets talk about this in person.